src/main/java/spoon/support/reflect/declaration/CtInterfaceImpl.java

```diff
@@ -109,4 +113,48 @@ public boolean isInterface() {
 	public String getLabel() {
 		return null;
 	}
+
+	@Override
+	public <N, C extends CtType<T>> C addNestedType(CtType<N> nestedType) {
+		super.addNestedType(nestedType);
+
+		if (nestedType == null) {
+			return (C) this;
+		}
+
+		// Type members of interfaces are implicitly public static. We need to add the implicit
+		// modifiers if they aren't public static already.
+		Set<CtExtendedModifier> modifiers = new HashSet<>(nestedType.getExtendedModifiers());
+		if (!nestedType.isPublic()) {
+			modifiers.add(new CtExtendedModifier(ModifierKind.PUBLIC, true));
+		}
+		if (!nestedType.isStatic()) {
+			modifiers.add(new CtExtendedModifier(ModifierKind.STATIC, true));
+		}
+		nestedType.setExtendedModifiers(modifiers);
+
+		return (C) this;
+	}
+
+	@Override
+	public <N> boolean removeNestedType(CtType<N> nestedType) {
+		if (!super.removeNestedType(nestedType)) {
+			return false;
+		}
+
+		// We might have added implicit public static modifiers so we need to remove them again
+		Set<CtExtendedModifier> newModifiers = nestedType.getExtendedModifiers().stream()
+				.filter(modifier -> {
+					if (!modifier.isImplicit()) {
+						return true;
+					}
+					return modifier.getKind() != ModifierKind.STATIC
+							&& modifier.getKind() != ModifierKind.PUBLIC;
+				})
+				.collect(Collectors.toCollection(HashSet::new));
```

Comment:
Minor Style comment: convert the filter in 2 filter statements, which are shorter and don't need `{... }` brackets

Reply:
That has different semantics but if you got confused and I needed a double-take, this code has to go. I will rewrite that with a `removeIf` later :)